app/main.ipynb

```diff
@@ -126,19 +163,46 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
-   "outputs": [],
    "source": [
+    "from itertools import combinations\n",
+    "\n",
+    "\n",
     "array = np.array([2.3, 7.8, 3.2, 1.1, 5.8, 9.5, 17.6, 11.1])\n",
-    "# write your code here"
+    "\n",
+    "target_sum = 27\n",
+    "\n",
+    "result = None\n",
+    "for r in range(1, len(array) + 1):\n",
+    "    for combo in combinations(array, r):\n",
+    "        if sum(combo) == target_sum:\n",
+    "            result = combo\n",
+    "            break\n",
+    "    if result:\n",
+    "        break\n",
+    "\n",
+    "\n",
+    "array[[0, 1, 4, -1]].sum()"
```

Comment:
In Task 4, the use of `itertools.combinations` to find a subsequence that sums to 27 is unnecessary. The task can be solved directly by using multi-indexing to select the desired elements from the array.